libsolidity/experimental/analysis/TypeInference.cpp

```diff
@@ -772,12 +791,21 @@ bool TypeInference::visit(TypeDefinition const& _typeDefinition)
 		return false;
 
 	if (_typeDefinition.arguments())
+	{
+		ScopedSaveAndRestore expressionContext{m_expressionContext, ExpressionContext::Type};
 		_typeDefinition.arguments()->accept(*this);
+	}
```

Comment:
I just noticed that the type variables in a type definition were not being visited in type context. This seems incorrect to me but still happened to work because `visit(VariableDeclaration)` initializes the type of a term with a fresh type variable. 

Now that visitation in type context will be a separate phase in my upcoming PR, it became a problem.